provider/azure-devops-workitems/api.ts

```diff
@@ -0,0 +1,202 @@
+    State: 'System.State',
+    Tags: 'System.Tags',
+}
+
+export interface MinimalWorkItem {
+    id: string
+    title: string
+    description: string
+    url: string
+}
+
+export interface SimpleWorkItem {
+    id: string
+    url: string
+    fields: {
+        title: string
+        state: string
+        type: string
+        assignedTo: string
+        description: string
+        tags: string[]
+    }
+}
+
+const authHeaders = (settings: Settings) => ({
+    Authorization: `Basic ${Buffer.from(`:${settings.accessToken}`).toString('base64')}`,
+    'content-type': 'application/json',
+})
+
+const buildUrl = (settings: Settings, path: string, searchParams: Record<string, string> = {}) => {
+    // Avoid double / if settings.url ends with '/' and path starts with '/'
+    const url = new URL(settings.url.replace(/\/$/, '') + path)
+    url.search = new URLSearchParams(searchParams).toString()
+    return url
+}
+
+export const searchWorkItems = async (
+    query: string | undefined,
+    settings: Settings
+): Promise<MinimalWorkItem[]> => {
+    let searchId = Number(query)
+    if (Number.isNaN(searchId)) {
+        searchId = 0
+    }
+    const wiql = query
+        ? `SELECT  [System.Id]
+FROM workitems
+WHERE ([System.Title] Contains '${query}' OR [System.Description] Contains '${query}' OR [System.Id] = ${searchId})
+ORDER BY [System.ChangedDate] DESC
+`
+        : `SELECT  [System.Id]
+FROM workitems
+WHERE ([System.AssignedTo] = @Me OR [System.CreatedBy] = @Me OR [System.ChangedBy] = @Me)
+ORDER BY [System.ChangedDate] DESC
+`
+
+    const queryUrl = buildUrl(settings, '/_apis/wit/wiql', {
+        'api-version': DEFAULT_API_VERSION,
+        $top: `${NUMBER_OF_ISSUES_TO_FETCH}`,
+    })
+    console.log(`Calling Azure DevOps API: ${queryUrl}`)
+    const queryResponse = await fetch(queryUrl, {
+        method: 'POST',
+        headers: authHeaders(settings),
+        body: JSON.stringify({
+            query: wiql,
+        }),
+    })
+    if (!queryResponse.ok) {
+        throw new Error(
+            `Error fetching Azure DevOps query (${queryResponse?.status} ${
+                queryResponse?.statusText
+            }): ${await queryResponse?.text()}`
+        )
+    }
+
+    const queryResult = (await queryResponse.json()) as WorkItemQueryResult
+
+    if (queryResult?.workItems?.length) {
+        // Query the ids and get the work items
+        const workItemIds = queryResult.workItems
+            .map(w => w.id)
+            .filter(id => id !== undefined) as number[]
+
+        const fields = [
+            WorkItemFields.Id,
+            WorkItemFields.Title,
+            WorkItemFields.WorkItemType,
+            WorkItemFields.Description,
+        ]
+        const wiUrl = buildUrl(settings, '/_apis/wit/workitems', {
+            'api-version': DEFAULT_API_VERSION,
+            fields: fields.join(','),
+            ids: workItemIds.join(','),
+        })
+        console.log(`Calling Azure DevOps API: ${wiUrl}`)
```

Comment:
if you wanna keep this log, rather add an optional logger field to your settings interface.